### PR TITLE
acme.sh: init at 2.7.8

### PR DESCRIPTION
**Pull request by yorickvP**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/admin/acme.sh/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    rev = version;
+    sha256 = "0zm64z7av63xi7yjhljab2i8q1vx4q1mpcmcm58jm6k4babalxrf";
+  };
+
+  phases = [ "installPhase" ];
+
+  installPhase = ''
+    mkdir -p $out $out/bin $out/lib
+    cp -R $src/* $_
+    ln -s $out/lib/acme.sh $out/bin
```
Comment (Mic92) [inline pkgs/tools/admin/acme.sh/default.nix]: Should be wrapped with `wrapProgram` to have openssl and curl in path: https://github.com/Neilpang/acme.sh/blob/master/acme.sh#L31

Comment (yegortimoshenko) [inline pkgs/tools/admin/acme.sh/default.nix]: I'm not sure if that belongs to `lib`: `share` subdirectory seems to be more appropriate. Generally, when upstream is not FHS-compliant, it is usually cleaner to just expose a symlink to executable:

```
  installPhase = "mkdir -p $out/bin && ln -s $src/acme.sh $_";
```

Comment (Mic92) [reply pkgs/tools/admin/acme.sh/default.nix]: Did you really meant to link from `$src` here?

Comment (yegortimoshenko) [reply pkgs/tools/admin/acme.sh/default.nix]: `$src` is just a fixed-output derivation, so it shouldn't be garbage collected if there are references to it. In this case, repo mostly contains documentation and the shell script itself. Is there a problem with this approach?

Comment (yegortimoshenko) [reply pkgs/tools/admin/acme.sh/default.nix]: I'm mostly worried about clutter and increased risk of collisions when it comes to `environment.pathsToLink`.

Comment (Mic92) [reply pkgs/tools/admin/acme.sh/default.nix]: Only it requires to pull two derivations instead of one, but maybe that's ok.

Comment (yegortimoshenko) [reply pkgs/tools/admin/acme.sh/default.nix]: There is also `libexec` that is not symlinked by default which also seems appropriate.

**Pull request merged**